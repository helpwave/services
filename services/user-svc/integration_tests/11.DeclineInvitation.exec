#!/bin/bash

echo "exists:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InJlamVjdG9yQGhlbHB3YXZlLmRlIiwibmFtZSI6IlRlc3RpbmUgVGVzdCIsIm5pY2tuYW1lIjoidGVzdGluZS50ZXN0Iiwib3JnYW5pemF0aW9ucyI6WyIzYjI1YzZmNS00NzA1LTQwNzQtOWZjNi1hNTBjMjhlYmE0MDYiXX0=" \
	-d '{"invitation_id": "0bdd571f-54d8-42f6-b449-624cfdead279"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/DeclineInvitation

echo "exists, not pending:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InJlamVjdG9yMkBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"invitation_id": "3232a80e-181f-462f-ad0d-8db54881afa7"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/DeclineInvitation 2>&1 || echo ""

echo "does not exist:"
grpcurl --plaintext -H "X-Organization: 3b25c6f5-4705-4074-9fc6-a50c28eba406" \
	-H "Authorization: Bearer eyJzdWIiOiIxODE1OTcxMy01ZDRlLTRhZDUtOTRhZC1mYmI2YmIxNDc5ODQiLCJlbWFpbCI6InJlamVjdG9yMkBoZWxwd2F2ZS5kZSIsIm5hbWUiOiJUZXN0aW5lIFRlc3QiLCJuaWNrbmFtZSI6InRlc3RpbmUudGVzdCIsIm9yZ2FuaXphdGlvbnMiOlsiM2IyNWM2ZjUtNDcwNS00MDc0LTlmYzYtYTUwYzI4ZWJhNDA2Il19" \
	-d '{"invitation_id": "3232a80e-181f-462f-ad0d-8db54881afa2"}' \
	localhost:3002 \
	proto.services.user_svc.v1.OrganizationService/DeclineInvitation 2>&1 || echo ""
