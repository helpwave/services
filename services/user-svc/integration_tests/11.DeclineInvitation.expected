exists:
{}
exists, not pending:
ERROR:
  Code: InvalidArgument
  Message: only pending invitations can be rejected

does not exist:
ERROR:
  Code: InvalidArgument
  Message: record not found

